### PR TITLE
Refactor `BaseActiveRecord::getDirtyAttributes()`

### PR DESCRIPTION
**Pull request by Tigrov**
* Improve performance
* Allow to use class properties (mixed with `$attributes` if no class properties) for table row values.

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/BaseActiveRecord.php [hunk 1]
```diff
@@ -135,21 +140,14 @@ public function getAttribute(string $name): mixed
...
         if ($except !== []) {
             $names = array_diff($names, $except);
         }
 
-        foreach ($names as $name) {
-            $values[$name] = $this->$name;
-        }
-
-        return $values;
+        return array_intersect_key($attributes, array_flip($names));
```
Comment (xepozz) [inline src/BaseActiveRecord.php:150]: Could you bench it?

Comment (xepozz) [reply src/BaseActiveRecord.php:150]: What about `oldAttributes`?

Comment (Tigrov) [reply src/BaseActiveRecord.php:150]: > Could you bench it?

Before:
`benchGetAttributes......................R1 I1 - Mo20.920μs (±1.00%)`

After:
`benchGetAttributes......................R2 I2 - Mo17.238μs (±1.78%)`

Comment (Tigrov) [reply src/BaseActiveRecord.php:150]: > What about `oldAttributes`?

`getOldAttributes()` is a different method, without changes.

**Review by darkdef — APPROVED** [review]

**Pull request merged**